jsapp/xlform/src/model.base.coffee

```diff
@@ -107,7 +115,16 @@ module.exports = do ->
       # In the quick fix, this is done in the view for 'required' rowDetails
       # (grep: XLF.configs.truthyValues)
 
-      if value not in [undefined, false, null]
```

Comment:
`_innerValue(...)` was necessary because in certain cirtumstances, value is set to `{"value":false, "_hideUnlessChanged": true}`, which would evaluate to `false`.

I also found that changing the order of the "if" conditions caught an edge case, and didn't change behavior negatively.